### ISSUE TITLE
Casting to NUMERIC from TEXT returns wrong results

### ISSUE THREAD

**Issue by proddata** [crate/crate#16647]
### CrateDB version

5.8.1 / 5.8.2

### CrateDB setup information

3-node CrateDB Cloud Cluster

> [!WARNING]
> Only happens in multi-node environments

### Problem description

Casting to NUMERIC from TEXT returns wrong results

This seems to be similar to: https://github.com/crate/crate/issues/15279

### Steps to Reproduce

```sql
DROP TABLE IF EXISTS t01;
CREATE TABLE t01 (txt TEXT);

INSERT INTO t01
VALUES ('22.10'), ('22.20'), ('22.40');

REFRESH TABLE t01;

SELECT
  txt,
  txt::DECIMAL,
  CAST(txt AS DECIMAL)
FROM
  t01;
```

### Actual Result

<img width="800" alt="image" src="https://github.com/user-attachments/assets/331dc064-1991-41cd-a4f8-a551b48790db">
(multi-node) 

### Expected Result

<img width="800" alt="image" src="https://github.com/user-attachments/assets/7e56db47-908e-42ab-821f-9b67b4dc55bb">
(single-node) 

### Workaround

Unfortunately there doesn't seem to be a trivial workaround, without lose of precision (e.g. casting to `DOUBLE`)

```sql
SELECT
  txt,
  txt::DOUBLE,
  CAST(txt AS DOUBLE)
FROM
  t01;
```

**Comment by romseygeek** [comment]
It looks like a bug in the Streamer implementation for NumericType.  This fails:
```
    @Test
    public void test_decimal_value_streaming() throws IOException {
        BigDecimal expected = NumericType.INSTANCE.implicitCast("20.24");
        BytesStreamOutput out = new BytesStreamOutput();
        NumericType.INSTANCE.writeValueTo(out, expected);
        StreamInput in = out.bytes().streamInput();
        BigDecimal actual = NumericType.INSTANCE.readValueFrom(in);
        assertThat(actual).isEqualTo(expected);
    }
```

```
org.opentest4j.AssertionFailedError: 
expected: 20.24
 but was: 2024
```

If we use a NumericType with a defined scale and precision (i.e. replace `NumericType.INSTANCE` with `new NumericType(10,2)`) then the test passes.

**Comment by romseygeek** [comment]
So for a workaround you should be able to cast using a specific scale and precision: `SELECT txt::NUMERIC(10,2)`

**Comment by seut** [comment]
@proddata Thank you for reporting. The fix will be released with the next `5.8.4` hotfix release.
